assert/assertions.go

```diff
@@ -301,13 +301,24 @@ func messageFromMsgAndArgs(msgAndArgs ...interface{}) string {
 func indentMessageLines(message string, longestLabelLen int) string {
 	outBuf := new(bytes.Buffer)
 
-	for i, scanner := 0, bufio.NewScanner(strings.NewReader(message)); scanner.Scan(); i++ {
-		// no need to align first line because it starts at the correct location (after the label)
-		if i != 0 {
-			// append alignLen+1 spaces to align with "{{longestLabel}}:" before adding tab
+	msgScanner := bufio.NewScanner(strings.NewReader(message))
+
+	// a buffer is set manually to store the tokenization of the message string
+	// so that we can re-use the same buffer for each line of the message without
+	// any additional allocations. We set the maximum buffer length to 1 more
+	// than the length of the message to avoid exceeding the default
+	// MaxScanTokenSize while scanning lines. This can happen when there is a
+	// single very long line. Refer to issue #1525
+	msgScanner.Buffer(nil, len(message)+1)
+
+	first := true
+	for msgScanner.Scan() {
+		if !first {
 			outBuf.WriteString("\n\t" + strings.Repeat(" ", longestLabelLen+1) + "\t")
 		}
-		outBuf.WriteString(scanner.Text())
+
+		outBuf.Write(msgScanner.Bytes())
+		first = false
```

Comment:
Move this line in the `if !first` block as it will have no effect on the subsequent iterations.